tests/ztest/Kconfig

```diff
@@ -47,3 +47,11 @@ config ZTEST_PARAMETER_COUNT
 	default 1
 	help
 	  Maximum amount of concurrent return values / expected parameters.
+
+config ZTEST_INIT_APP_SHARED_MEM
```

Comment:
Ugh, I don't like that this config needs to exist.
We really need initialization of domains/partitions to be automatic at boot, like we do with statically defined kernel objects, and the only manual calls that need to be made are to assign threads to them.
I guess this can be addressed later.

Reply:
By deafult this is set to true. for instance if someone wants to have a test case in which they would like to manually do smem init, they can set this to No. This is just a extension people can use later.